src/Metadata/Resource/Factory/XmlResourceMetadataFactory.php

```diff
@@ -0,0 +1,239 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Metadata\Resource\Factory;
+
+use ApiPlatform\Core\Annotation\Resource;
+use ApiPlatform\Core\Exception\ResourceClassNotFoundException;
+use ApiPlatform\Core\Metadata\Resource\ResourceMetadata;
+
+/**
+ * Creates a resource metadata from xml {@see Resource} configuration.
+ *
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ */
+final class XmlResourceMetadataFactory implements ResourceMetadataFactoryInterface
+{
+    private $xmlParser;
+    private $paths;
+    private $decorated;
+
+    const RESOURCE_SCHEMA = __DIR__.'/../../../Bridge/Symfony/Bundle/Resources/schema/xml_metadata.xsd';
+
+    /**
+     * @param string[]                              $paths
+     * @param ResourceMetadataFactoryInterface|null $decorated
+     */
+    public function __construct(array $paths, ResourceMetadataFactoryInterface $decorated = null)
+    {
+        $this->xmlParser = new \DOMDocument();
+        $this->paths = $paths;
+        $this->decorated = $decorated;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create(string $resourceClass) : ResourceMetadata
+    {
+        $parentResourceMetadata = null;
+        if ($this->decorated) {
+            try {
+                $parentResourceMetadata = $this->decorated->create($resourceClass);
+            } catch (ResourceClassNotFoundException $resourceNotFoundException) {
+                // Ignore not found exception from decorated factories
+            }
+        }
+
+        try {
+            $reflectionClass = new \ReflectionClass($resourceClass);
+        } catch (\ReflectionException $reflectionException) {
+            return $this->handleNotFound($parentResourceMetadata, $resourceClass);
+        }
+
+        $metadata = null;
+
+        foreach ($this->paths as $path) {
+            $this->xmlParser->loadXML(file_get_contents($path));
+
+            if (false === @$this->xmlParser->schemaValidate(self::RESOURCE_SCHEMA)) {
+                throw new \InvalidArgumentException(sprintf('XML Schema loaded from path %s is not valid!', realpath($path)));
+            }
```

Comment:
As far as I can tell, Symfony does it everywhere too:

https://github.com/symfony/symfony/blob/af9ec74edb2498dc9eaf175e7e62815a06c8d6ba/src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php#L487

https://github.com/symfony/symfony/blob/f29d46f29b91ea5c30699cf6bdb8e65545d1dd26/src/Symfony/Component/Translation/Loader/XliffFileLoader.php#L173